cdc/processor/sinkmanager/redo_cache.go

```diff
@@ -81,6 +81,9 @@ func (r *redoEventCache) pop(
 			}
 			return pos.Compare(upperBound[0]) > 0
 		})
+		if fetchCount == 0 {
```

Comment:
Do we need to add a test for this change?

Reply:
done.